crates/hotshot/src/lib.rs

```diff
@@ -755,9 +755,11 @@ impl<TYPES: NodeType> HotShotInitializer<TYPES> {
     /// initialize from genesis
     /// # Errors
     /// If we are unable to apply the genesis block to the default state
-    pub fn from_genesis() -> Result<Self, HotShotError<TYPES>> {
+    pub fn from_genesis(
+        instance_state: &TYPES::InstanceState,
```

Comment:
Not a big deal but if we're passing in the instance state here, and this initializer ultimately gets passed to `SystemContext::init`, we might as well store the instance state in `HotShotInitializer` and remove it as a param from `init`/`new`